### PR TITLE
temporarily disable MinGW builds

### PR DESCRIPTION
**Pull request by mykmelez**
Temporarily disable MinGW builds due to https://github.com/rust-lang/rust/issues/47048 to fix bustage on Appveyor. Once that issue is fixed, presumably by https://github.com/rust-lang/rust/pull/51989, we should reenable them.

### CODE REVIEW

**Comment by mati865** [comment]
Landing of rust-lang/rust#51989 is facing many issues and can take a while.
Meanwhile you can try this workaround: https://github.com/nabijaczleweli/cargo-update/issues/74#issuecomment-376581009

File: .appveyor.yml [hunk 1]
```diff
@@ -12,6 +12,11 @@ install:
   - rust-nightly-%TARGET%.exe /VERYSILENT /NORESTART /DIR="C:\Program Files (x86)\Rust"
   - set PATH=%PATH%;C:\Program Files (x86)\Rust\bin
   - if defined MSYS_BITS set PATH=%PATH%;C:\msys64\mingw%MSYS_BITS%\bin
+  -
+  # Work around undefined references to __acrt_iob_func() per
+  # https://github.com/nabijaczleweli/cargo-update/issues/74#issuecomment-376581009
+  - sed -rie "s/#define std([[:alpha:]]+)[[:space:]]+\(__acrt_iob_func\(([[:digit:]]+)\)\)/#define std\1 (\&__iob_func()[\2])/" "C:\msys64\mingw32\i686-w64-mingw32\include\stdio.h" "C:\msys64\mingw64\x86_64-w64-mingw32\include\stdio.h"
```
Comment (mati865) [inline .appveyor.yml]: You have to use sed with `-s` to edit multiple files.

Comment (mykmelez) [reply .appveyor.yml]: > You have to use sed with -s to edit multiple files.

Thanks for the tip! Unfortunately, I'm still seeing the same error.

File: .appveyor.yml [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
     - TARGET: x86_64-pc-windows-gnu
       MSYS_BITS: 64
-    - TARGET: i686-pc-windows-gnu
-      MSYS_BITS: 32
 
 install:
   - ps: Start-FileDownload "https://static.rust-lang.org/dist/rust-nightly-${env:TARGET}.exe"
   - rust-nightly-%TARGET%.exe /VERYSILENT /NORESTART /DIR="C:\Program Files (x86)\Rust"
   - set PATH=%PATH%;C:\Program Files (x86)\Rust\bin
-  - if defined MSYS_BITS set PATH=%PATH%;C:\msys64\mingw%MSYS_BITS%\bin
```
Comment (mati865) [inline .appveyor.yml]: MSYS2 GCC needs `__acrt_iob_func` but Rust shipped GCC doesn't like it, try without this line.

**Comment by mati865** [comment]
Now I'm confused, Rust used to ship it's own GCC build.
Anyway try with original `- if defined MSYS_BITS set PATH=%PATH%;C:\msys64\mingw%MSYS_BITS%\bin` I'll see later today after I get some sleep.

**Comment by mati865** [comment]
Sorry for the delay, I had no time to work on it.
This workaround doesn't work because there `__acrt_iob_func()` is being used in the other places and this project uses more C/C++ code than `cargo-update`.
It's possible to replace it everywhere but it could break build it in another way.

**Comment by mykmelez** [comment]
> Sorry for the delay, I had no time to work on it.
> This workaround doesn't work because there __acrt_iob_func() is being used in the other places and this project uses more C/C++ code than cargo-update.
> It's possible to replace it everywhere but it could break build it in another way.

No worries, thanks for looking at it! I'll keep an eye out for other workarounds and perhaps land the original commit that disables the MinGW builds in the meantime.

**Pull request merged**